src/BoundsInference.cpp

```diff
@@ -109,27 +110,120 @@ class BoundsInference : public IRMutator {
     set<string> in_pipeline, inner_productions;
     Scope<int> in_stages;
 
+    struct CondValue {
+        Expr cond; // Condition on params only (can't depend on loop variable)
+        Expr expr;
```

Comment:
The values inside 'exprs' vector can be of type 'Stmt' or 'Expr' depending on whether they have RDom predicates associated with them or not. Template wouldn't work in this case.